### ISSUE TITLE
Check of code approach

### ISSUE THREAD

**Issue by mwhite192** [mwhite192/P5-Web-Dev-Kanap#4]
@AccessiT3ch - Hey Conor!  I was able to get both the sum of the items and total price of the cart added to the DOM!  I have to admit it was really fun figuring things out! :)  

I do have a request though.  I know in our last session you stated that it was best to write functions, so all I had to do was call them in the event listeners.  Was this statement pertaining to just the event listeners (meaning the action that's suppose to be performed in the event listeners) or is it best to write function whenever possible?  I ask because I didn't write functions to populate the sum of articles or the total of the cart (unless Promise.all is a function in its own right).  I wanted to make sure I did the right thing.

If it was okay that I didn't write function for these two items, would you be so kind as to check my code to make sure that is consolidated and I did things 'correctly' as in 'this works but its not best practice to do this' lol  :) please and thank you?

**Comment by AccessiT3ch** [comment]
Hey @mwhite192 - Sorry, I thought I replied to this but clearly I didn't come back to it!

We don't have to write functions for every little thing, but we certainly can, so it's definitely a bit of a balance between function modularity / readability and the overhead of defining tons of functions. 

One way I like to think about it is from a testing point of view: every function has either an output (something that's returned) and/or an outcome (like updating the DOM), can that functionality be programmatically tested (via unit / integration tests)? If not, it may be risky code and/or hard-to-debug code. 

Another aspect is just how much functionality any one function is responsible for, big functions that do a lot of things are fine as long as their manageable and fully testable. If you find that you're repeating a bit of functionality, pulling even a single line of code into its own function can help DRY your code.

**Comment by AccessiT3ch** [comment]
In general your functions have been looking well scoped to me, I'll definitely take a look though!

**Comment by AccessiT3ch** [comment]
@mwhite192 In your cart script it looks like you're fetching from the API multiple times (for the same data), instead you could return the data at the end of your `.then()` to save the data to a variable.

```js
const cartData = Promise.all(...).then((data) => {
  // everything else in here can remain the same
  return data;
});
```

Alternatively, you could chain both your `.then()` functions on the same Promise.all().

```js
Promse.all(....).then((data) => {
   // customer order functionality
  return data;
}).then((data) => {
  // cart total functionality
});
```

Or you could do both sets of functionality in the same `.then()`.
```js
Promise.all(...).then((data) => {
  // customer order functionality
  // cart total functionality
});
```
Ideally, we make as few requests to the API as possible!

**Comment by AccessiT3ch** [comment]
@mwhite192 do you have an event listener on the Order button on the cart page? Looks like you still need to validate the order form and then POST the data to the order product route/controller.

- [ ] Validate the email is an email with regex
- [ ] Validate that all other form inputs are not empty
- [ ] Use `fetch()` to POST the order data to the API
- [ ] Use `.then` (after fetch) to navigate to the confirmation page

**Comment by mwhite192** [comment]
@AccessiT3ch - Hey Conor!  Thank you so much for this information!  I was trying to find a way to keep my code DRY since I requested the same data from the API twice.  I decided to do the same functionality in one .then() because that kept my code clean and it made the most sense.  

I have still stuck on the event listener actually.  I am having trouble trying to set the event listener on the "item quantity" element.  I have tried everything and haven't been able to get the element.  Every way I code I get a 'null' result.  The milestone document says to use element.closest(), but that hasn't worked for me either.
